nf_core/modules/module_test.py

```diff
@@ -71,25 +85,66 @@ def run(self):
     def _check_inputs(self):
         """Do more complex checks about supplied flags."""
 
+        # Retrieving installed modules
+        if self.repo_type == "modules":
+            local = False
+            installed_modules = self.module_names["modules"]
+        else:
+            local = questionary.confirm("Is the module local?", style=nf_core.utils.nfcore_question_style).unsafe_ask()
+            if local:
+                installed_modules = self.module_names.get("local/modules")
+            else:
+                installed_modules = self.module_names.get("nf-core/modules")
+
         # Get the tool name if not specified
         if self.module_name is None:
             if self.no_prompts:
                 raise UserWarning(
                     "Tool name not provided and prompts deactivated. Please provide the tool name as TOOL/SUBTOOL or TOOL."
                 )
-            modules_repo = ModulesRepo()
-            modules_repo.get_modules_file_tree()
+            if installed_modules is None:
+                raise UserWarning(
+                    "No installed modules were found from '{self.modules_repo.remote_url}'.\n"
+                    "Are you running the tests inside the nf-core/modules main directory?\n"
+                    "Otherwise, make sure that the directory structure is modules/TOOL/SUBTOOL/ and tests/modules/TOOLS/SUBTOOL/"
+                )
             self.module_name = questionary.autocomplete(
                 "Tool name:",
-                choices=modules_repo.get_avail_modules(),
+                choices=installed_modules,
                 style=nf_core.utils.nfcore_question_style,
             ).unsafe_ask()
-        module_dir = Path("modules") / self.module_name
 
-        # First, sanity check that the module directory exists
-        if not module_dir.is_dir():
+        # Sanity check that the module directory exists
+        self._validate_folder_structure(local)
+
+    def _validate_folder_structure(self, local=False):
+        """Validate that the modules follow the correct folder structure to run the tests:
+        - modules/TOOL/SUBTOOL/
+        - tests/modules/TOOL/SUBTOOL/
+
+        local (bool): Testing local modules (True) or nf-core modules (False)
+        """
+        if local:
+            basedir = "modules/local"
+        else:
+            basedir = "modules/nf-core"
+
+        if self.repo_type == "modules":
+            module_path = Path("modules") / self.module_name
+            test_path = Path("tests/modules") / self.module_name
+        else:
+            module_path = Path(f"{basedir}/modules") / self.module_name
```

Comment:
This fails for local modules since they are all kept as plain `.nf` files in the `modules/local` directory of a pipeline.

Reply:
To run the tests you must have the same folder structure that inside modules/nf-core, I think it's correct to make it fail if the structure is not the same, otherwise there wouldn't be tests to run.

Reply:
Actually, it may be better to change the error message below to make it clear

Reply:
I'm wondering the if it is worth the hassle to adapt the commands to local modules, will anyone use it?

Reply:
Then I'd say we skip it